basics/data-import/pinot-stream-ingestion/configure-indexes.md

```diff
@@ -0,0 +1,175 @@
+# Configure indexes
+
+Learn how to apply indexes to a Pinot table. This guide assumes that you have followed the [Ingest data from Apache Kafka](/basics/data-import/pinot-stream-ingestion/import-from-apache-kafka.md) guide.
+
+Pinot supports a series of different indexes that can be used to optimise query performance. In this guide we'll learn
+how to add indexes to the `events` table that we set up in the [Ingest data from Apache Kafka](/basics/data-import/pinot-stream-ingestion/import-from-apache-kafka.md) guide.
+
+### Why do we need indexes?
+
+If no indexes are applied to the columns in a Pinot segment, the query engine needs to scan through every document,
+checking whether that document meets the filter criteria provided in a query. This can be a slow process if there are a
+lot of documents to scan.
+
+When indexes are applied, the query engine can more quickly work out which documents satisfy the filter criteria,
+reducing the time it takes to execute the query.
+
+### What indexes does Pinot support?
+
+By default, Pinot creates a forward index for every column. The forward index generally stores documents in insertion
+order.
+
+However, before flushing the segment, Pinot does a single pass over every column to see whether the data is sorted. If
+it is then Pinot instead creates a sorted (forward) index for that column.
+
+<Callout>
+  For real-time tables you can also explicitly tell Pinot that one of the columns should be sorted. For more details,
+  see the [Sorted Index Documentation](https://docs.pinot.apache.org/basics/indexing/forward-index#real-time-tables).
+</Callout>
+
+For filtering documents within a segment, Pinot supports the following indexing techniques:
+
+* Inverted index - Used for exact lookups
+* Range index - Used for range queries.
+* Text index - Used for phrase, term, boolean, prefix, or regex queries.
+* Geospatial index - Based on H3, a hexagon-based hierarchical gridding. 
+Used for finding points that exist within a certain distance from another point.
+* JSON index - Used for querying columns in JSON documents.
+* Star-Tree index - Pre-aggregates results across multiple columns. 
+
+## View events table
+
+Let's see how we can apply these indexing techniques to our data.
+To recap, the `events` table has the following fields:
+
+| Date Time Fields       | Dimensions Fields | Metric Fields |
+|----------|-------------| -------------|
+| `ts` |`uuid` |     `count` |
+
+We might want to write queries that filter on the `ts` and `uuid` columns, so these are the columns on which we would want to configure indexes.
+
+Since the data we're ingesting into the Kafka topic is all implicitly ordered by timestamp, this means that the `ts` column already has a sorted index.
+This means that any queries that filter on this column are already optimised.
```

Comment:
```suggestion
Because the data we're ingesting into the Kafka topic is implicitly ordered by timestamp, the `ts` column already has a sorted index. Having a sorted index means queries that filter on this column are already optimized.
```